multibase.csv

```diff
@@ -1,26 +1,26 @@
-base64url,         u,    rfc4648 no padding,                                           default
-base64urlpad,      U,    rfc4648 with padding,                                         default
-proquint,          p,    PRO-QUINT https://arxiv.org/html/0901.4016,                   draft
-base256emoji,      🚀,    base256 with custom alphabet using variable-sized-codepoints, draft
+identity,          0x00 (NUL),  8-bit binary (encoder and decoder keeps data unmodified),     default
+base2,             0x30 (0),    binary (01010101),                                            candidate
+base8,             0x37 (7),    octal,                                                        draft
+base10,            0x39 (9),    decimal,                                                      draft
+base16,            0x66 (f),    hexadecimal,                                                  default
+base16upper,       0x46 (F),    hexadecimal,                                                  default
+base32hex,         0x76 (v),    rfc4648 case-insensitive - no padding - highest char,         candidate
+base32hexupper,    0x56 (V),    rfc4648 case-insensitive - no padding - highest char,         candidate
+base32hexpad,      0x74 (t),    rfc4648 case-insensitive - with padding,                      candidate
+base32hexpadupper, 0x54 (T),    rfc4648 case-insensitive - with padding,                      candidate
+base32,            0x62 (b),    rfc4648 case-insensitive - no padding,                        default
+base32upper,       0x42 (B),    rfc4648 case-insensitive - no padding,                        default
+base32pad,         0x63 (c),    rfc4648 case-insensitive - with padding,                      candidate
+base32padupper,    0x43 (C),    rfc4648 case-insensitive - with padding,                      candidate
+base32z,           0x68 (h),    z-base-32 (used by Tahoe-LAFS),                               draft
+base36,            0x6b (k),    base36 [0-9a-z] case-insensitive - no padding,                draft
+base36upper,       0x4b (K),    base36 [0-9a-z] case-insensitive - no padding,                draft
+base58btc,         0x7a (z),    base58 bitcoin,                                               default
+base58flickr,      0x5a (Z),    base58 flicker,                                               candidate
+base64,            0x6d (m),    rfc4648 no padding,                                           default
+base64pad,         0x4d (M),    rfc4648 with padding - MIME encoding,                         candidate
+base64url,         0x75 (u),    rfc4648 no padding,                                           default
+base64urlpad,      0x55 (U),    rfc4648 with padding,                                         default
+proquint,          0x70 (p),    PRO-QUINT https://arxiv.org/html/0901.4016,                   draft
+base256emoji,      0xe7 (🚀),    base256 with custom alphabet using variable-sized-codepoints, draft
```

Comment:
This should probably be 2 rows 

```
base256emoji, 0xF09F9A80 (🚀) utf-8 base256 with custom alphabet using variable-sized-codepoints, draft
base256emoji, 0xD83DDE80 (🚀) utf-16 base256 with custom alphabet using variable-sized-codepoints, draft
```

UTF-32 would be problematic as it begins with 0x00 and is indistinguishable from raw `0x0001F680` is a valid multibase for identity `[01, F6, 80]`

Encoding | Bytes
-- | --
UTF-8 | 0xF0 0x9F 0x9A 0x80
UTF-16 | 0xD83D 0xDE80
UTF-32 | 0x0001F680

A parser should assume that if the first byte is `0xF0` or `0xD8` that it is Unicode and needs to look at more bytes to determine the actual Unicode encoding being used and if it is base256emoji.

also, someone should check my endianness

Reply:
a great argument against the UTF-8 column! hehe